src/credentials/CHIPCert.h

```diff
@@ -195,15 +197,18 @@ struct ChipDN
         uint64_t mChipId; /**< CHIP specific DN attribute value. */
         struct
         {
-            const uint8_t * mValue; /**< Pointer to the DN attribute value. */
-            uint32_t mLen;          /**< DN attribute length. */
-        } mString;                  /**< DN attribute structure when encoded as a string. */
-    } mAttrValue;                   /**< DN attribute value union: string or unsigned integer. */
-    chip::ASN1::OID mAttrOID;       /**< DN attribute CHIP OID. */
+            const uint8_t * mValue;    /**< Pointer to the DN attribute value. */
+            uint32_t mLen;             /**< DN attribute length. */
+        } mString;                     /**< DN attribute structure when encoded as a string. */
+    } mAttrValue[RDN_NUM];             /**< DN attribute value union: string or unsigned integer. */
+    chip::ASN1::OID mAttrOID[RDN_NUM]; /**< DN attribute CHIP OID. */
```

Comment:
some comments about invariants may be useful.

It seems that we enforce that kOID_AttributeType_ChipNodeId is always index 0 and any other are at higher indices.

Reply:
done

Reply:
If index 0 is always a certain thing, and indices 1+ are always different things, does it make sense to hold the index 0 thing in the array at all?  Why not give it its own dedicated field?

Reply:
This is not correct. The spec doesn't define specific order of Attributes in the DN. So kOID_AttributeType_ChipNodeId attribute can appear at index 0, 1, 2...

Reply:
OK, I understand.  Then yes, the array is definitely the correct structure because (as discussed elsewhere) order matters for the purposes of comparison AND order is not prescribed by the spec.  It would seem then that index 0 is _not_ special.  But order _is_ special and needs to be preserved.